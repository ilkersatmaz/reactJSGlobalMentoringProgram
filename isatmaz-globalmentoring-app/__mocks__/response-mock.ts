export const movieResponse = {
	data: {
		id: 221122,
		title: 'first movie of the year',
		overview:
			'overview of the video for the m ',		
		genres: ['Action', 'Horror'],
		budget: 1000,
		poster_path: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
		release_date: '2023-01-01',
		revenue: 136907000,
		runtime: 134,
		tagline: "new and best",
		
		vote_average: 9.9,
		vote_count: 1145,
	},

	totalAmount: 1,
};
