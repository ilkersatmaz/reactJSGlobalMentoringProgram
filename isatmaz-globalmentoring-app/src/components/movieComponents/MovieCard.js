import React, { useState } from "react";
import './Movie.css'
import Popup from 'reactjs-popup';
import { BsThreeDotsVertical } from 'react-icons/bs';
import UpsertMovie from "../movieComponents/upsertMovieComponent/UpsertMovie";
import DeleteMovie from "./deleteMovieComponent/DeleteMovie";

function MovieCard(props){
    const {movieData, onMovieClick} = props;
    const [isMoreOpen, setIsMoreOpen]=useState(false);
    const [isUpsertOpen, setIsUpsertOpen]=useState(false);
    function getYear(releaseDate){
        if(releaseDate){
            const date=new Date(releaseDate)
            return date.getFullYear();
        }
        return "";
    }
    return (
        <div className="movie-card-div" onClick={()=>{onMovieClick(movieData)}}>
            <button className="movie-card-more-button" onClick={()=>{setIsMoreOpen(!isMoreOpen)}}><BsThreeDotsVertical color="#FFFFFF" className="movie-card-more-icon"/></button>
            <div className={isMoreOpen ? `more-option-div` : `more-option-div-hide`} >
                <div>
                    <Popup className="upsert-popup" onOpen={()=>{setIsUpsertOpen(true); setIsMoreOpen(false)}} open={isUpsertOpen} trigger={<button className="movie-card-edit-button"> <p className="more-option-text">Edit</p> </button>} position="bottom left" arrow={false}>
                        <UpsertMovie movieData={movieData} setIsUpsertOpen={setIsUpsertOpen} />
                    </Popup>   
                    <Popup className="upsert-popup" trigger={ <button className="movie-card-edit-button"> <p className="more-option-text">Delete</p></button>} position="bottom left" arrow={false}>
                        <DeleteMovie movieData={movieData} />
                    </Popup> 
                </div>
            </div>

            <img className="movie-card-image" alt={movieData.title} src={movieData.imageUrl} />
            <div className="movie-card-title-div">
                <p className="movie-card-title">{movieData.title}</p>                
                <p className="movie-card-releaseDate">{getYear(movieData.releaseDate)}</p>                
            </div>            
            <div className="movie-card-genre-div">
                {movieData.genre.map((genre, index)=> {
                    return <p key={genre} className="movie-card-genre">{genre}{ (index===movieData.genre.length-1 ? '' : ',')}</p>
                })}
            </div>   
        </div>        
    )
}

export default MovieCard;