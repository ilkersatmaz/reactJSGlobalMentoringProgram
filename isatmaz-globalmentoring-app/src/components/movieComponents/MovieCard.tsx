import React, { useState } from "react";
import './Movie.css'
import Popup from 'reactjs-popup';
import { BsThreeDotsVertical } from 'react-icons/bs';
import UpsertMovie from "./upsertMovieComponent/UpsertMovie";
import DeleteMovie from "./deleteMovieComponent/DeleteMovie";
import ConfirmationPopup from "./confirmationComponent/ConfirmationPopup";

function MovieCard(props:any){
    const {movieData, onMovieClick} = props;
    const [isMoreOpen, setIsMoreOpen]=useState(false);
    const [isUpsertOpen, setIsUpsertOpen]=useState(false);
    const [isConfirmationOpen, setIsConfirmationOpen]=useState(false);
    function getYear(releaseDate:string){
        if(releaseDate){
            const date=new Date(releaseDate)
            return date.getFullYear();
        }
        return "";
    }
    return (
        <div className="movie-card-div" onClick={()=>{onMovieClick(movieData)}}>
            <button className="movie-card-more-button" onClick={()=>{setIsMoreOpen(!isMoreOpen)}}><BsThreeDotsVertical color="#FFFFFF" className="movie-card-more-icon"/></button>
            <div className={isMoreOpen ? `more-option-div` : `more-option-div-hide`} >
                <div>
                    <Popup className="upsert-popup" onOpen={()=>{setIsUpsertOpen(true); setIsMoreOpen(false)}} open={isUpsertOpen} trigger={<button className="movie-card-edit-button"> <p className="more-option-text">Edit</p> </button>} position="bottom left" arrow={false}>
                        <UpsertMovie mode={"edit"} movieData={movieData} setIsUpsertOpen={setIsUpsertOpen} setIsConfirmationOpen={setIsConfirmationOpen} />
                    </Popup>   
                    <Popup className="upsert-popup" trigger={ <button className="movie-card-edit-button"> <p className="more-option-text">Delete</p></button>} position="bottom left" arrow={false}>
                        <DeleteMovie movieId={movieData.id} setIsConfirmationOpen={setIsConfirmationOpen}/>
                    </Popup>                           
                </div>
            </div>
            <Popup className="confirmation-popup" onClose={()=>{setIsConfirmationOpen(false)}} position="bottom left" open={isConfirmationOpen} arrow={false}>
                <ConfirmationPopup setIsConfirmationOpen={setIsConfirmationOpen}/>
            </Popup>  
            <img className="movie-card-image" alt={movieData.title} src={movieData.poster_path} />
            <div className="movie-card-title-div">
                <p className="movie-card-title">{movieData.title}</p>                
                <p className="movie-card-releaseDate">{getYear(movieData.release_date)}</p>                
            </div>            
            <div className="movie-card-genre-div">
                {movieData.genres.map((genre: string, index: number)=> {
                    return <p key={genre} className="movie-card-genre">{genre}{ (index===movieData.genres.length-1 ? '' : ',')}</p>
                })}
            </div>   
        </div>        
    )
}

export default MovieCard;