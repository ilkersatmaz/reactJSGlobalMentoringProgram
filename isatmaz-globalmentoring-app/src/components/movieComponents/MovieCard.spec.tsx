import { render } from '@testing-library/react';
import React from 'react';
import MovieCard from './MovieCard';
import { movieResponse } from '../../../__mocks__/response-mock';


describe('MovieCard component', () => {
  it("should match the snapshot", () => {
    const mockFn=jest.fn();
    const { asFragment } = render(<MovieCard movieData={movieResponse.data} onMovieClick={mockFn}/>);
    
    expect(asFragment).toMatchSnapshot();
  });
})