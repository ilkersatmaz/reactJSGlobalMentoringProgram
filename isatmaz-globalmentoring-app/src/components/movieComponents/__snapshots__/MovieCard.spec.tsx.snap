// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard component should match the snapshot 1`] = `[Function]`;
